### PR TITLE
feat: support custom `config.development.reloadPattern `

### PR DESCRIPTION
**Pull request by atian25**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

- support `config.development.reloadPattern `
```js
// config/config.default.js
exports.development = {
  // don't reload when ts fileChanged
  // https://github.com/sindresorhus/multimatch
  reloadPattern: ['**', '!**/*.ts'],
};
```
- close https://github.com/eggjs/egg/issues/2056

### CODE REVIEW

**Comment by atian25** [comment]
重构了下，直接支持 glob，暂不开放函数接口

File: README.md [hunk 1]
```diff
@@ -50,6 +50,17 @@ Under the following directory (including subdirectories) will ignore file change
...
 - ${app_root}/app/assets
 
+Developer can use `config.pattern`([multimatch](https://github.com/sindresorhus/multimatch)) to control whether to reload.
+
+```js
+// config/config.default.js
+exports.development = {
+  // don't reload when ts fileChanged
+  // https://github.com/sindresorhus/multimatch
+  pattern: ['**', '!**/*.ts'],
```
Comment (dead-horse) [inline README.md]: watchPattern?

Comment (atian25) [reply README.md]: 可以

Comment (atian25) [reply README.md]: 改为 reloadPattern 了，已提交

**Comment by dead-horse** [comment]
2.2.0

**Pull request merged**